tex/gregoriotex.lua

```diff
@@ -899,14 +899,15 @@ local function compile_gabc(gabc_file, gtex_file, glog_file, allow_deprecated)
         .."'%s' with %s.\nPlease check your score file.", gabc_file,
         gregorio_exe())
   else
-    -- open the gtex file for writing so that LuaTeX records output to it
-    -- when the -recorder option is used
-    local gtex = io.open(gtex_file, 'a')
-    if gtex == nil then
-      err("\n Unable to open %s", gtex_file)
-    else
-      gtex:close()
-    end
+    -- The next few lines would open the gtex file for writing so that LuaTeX records the fact that gregorio has written to it
+    -- when the -recorder option is used.
+    -- However, in restricted \write18 mode, the gtex file might not be writable. Since we're the sole consumer of the gtex file, it should be okay not to record the write.
+    --local gtex = io.open(gtex_file, 'a')
+    --if gtex == nil then
+    --  err("\n Unable to open %s", gtex_file)
```

Comment:
This is the line you're talking about, right?  What about simply converting the error to a warning?  Something to the effect that the `-recorder` option might not work correctly if you're making manual edits to the gtex file without the `--shell-escape` option.